### PR TITLE
Fp 2644: UI is frozen in certain split configurations in the ide

### PR DESCRIPTION
**Pull request by quirinpa**
- [FP-2644](https://movai.atlassian.net/browse/FP-2644): UI is frozen in certain split configurations in the IDE

[FP-2644]: https://movai.atlassian.net/browse/FP-2644?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/plugins/views/Tabs/hooks/useTabLayout.js [hunk 1]
```diff
@@ -138,9 +138,10 @@ const useTabLayout = (props, dockRef) => {
       let tabExists = maxboxChildren?.tabs.find((t) => t.id === layoutActiveId);
 
       if (!tabExists) {
-        tabExists = _layout.dockbox.children[0].tabs.find(
-          (t) => t.id === layoutActiveId,
-        );
+        if (_layout.dockbox.children[0].tabs)
```
[the remark this answers is not in the document]
Comment (quirinpa) [reply src/plugins/views/Tabs/hooks/useTabLayout.js:141]: It only alters the logic in the sense that it doesn't bug out if that object "tabs" is does not have a "find" method.

So I think you are incorrect about your pointing out of non-correctness.

It does not alter the logic at all.

[the remark this answers is not in the document]
Comment (manuelnogueiramov) [reply src/plugins/views/Tabs/hooks/useTabLayout.js:141]: Nevermind, it seemed like you replaced the previous if with the new one. But apparently you just added a new one inside it.

Good call :+1:

[the remark this answers is not in the document]
Comment (manuelnogueiramov) [reply src/plugins/views/Tabs/hooks/useTabLayout.js:141]: Only suggestion here would be to do:

```
const dockboxChildren = _layout.dockbox.children[0].tabs;
```

And use it in both places.

**Review by manuelnogueiramov — APPROVED** [review]
You were correct, I misread the code. Still left a simple suggestion, but it's not blocking. Good job :+1:

**Review by diasnad — APPROVED** [review]

**Comment by manuelnogueiramov** [comment]
The new code looks weird... where are we checking for the `dockbox` children now?

**Review by manuelnogueiramov — CHANGES_REQUESTED** [review]
The current code will introduce regressions.

**Comment by quirinpa** [comment]
You are incorrect, Manuel. The variable you suggested was already defined.

File: src/plugins/views/Tabs/hooks/useTabLayout.js [hunk 2]
```diff
@@ -140,9 +140,8 @@ const useTabLayout = (props, dockRef) => {
       let tabExists = maxboxChildren?.tabs.find((t) => t.id === layoutActiveId);
 
       if (!tabExists) {
-        tabExists = _layout.dockbox.children[0].tabs.find(
-          (t) => t.id === layoutActiveId,
-        );
+        if (maxboxChildren?.tabs)
+          tabExists = maxboxChildren.tabs.find((t) => t.id === layoutActiveId);
       }
       activeTabId.current = layoutActiveId;
```
Comment (manuelnogueiramov) [inline src/plugins/views/Tabs/hooks/useTabLayout.js:146]: I looked at this for some time, and this makes 0 sense. We're NEVER checking `dockbox` anymore for the tab. The first approach you did, seemed to work correctly. This is just nonsense. line 144 that you introduced is exactly the same as line 140. Where it used to be a check inside `dockbox`, not `maxbox` again.

**Review by manuelnogueiramov — CHANGES_REQUESTED** [review]

File: src/plugins/views/Tabs/hooks/useTabLayout.js (hunk 2 above)
Comment (quirinpa) [reply src/plugins/views/Tabs/hooks/useTabLayout.js:146]: Ok. In this case you are correct.

Comment (quirinpa) [reply src/plugins/views/Tabs/hooks/useTabLayout.js:146]: Testing it, it just works. Maybe that's due to the fact that the maxbox and the dockbox children are the same.
Do you have any reason to think of a situation where they aren't?

Comment (manuelnogueiramov) [reply src/plugins/views/Tabs/hooks/useTabLayout.js:146]: Testing what? Did you test every possible scenario? If you did and nothing broke, we're good to go. As I doubt that happened, my request changes still mantains. I know I added that a few years ago because of an edge case scenario. Approving this code as is will fix YOUR issue and cause at least 1 regression.
Before you ask, I don't remember what, but if you dig a bit you'll find the ticket that first originated that code and then you can test it.

Comment (quirinpa) [reply src/plugins/views/Tabs/hooks/useTabLayout.js:146]: Please, if you can think of more possible scenarios than I can. Help out testing.

**Comment by quirinpa** [comment]
I've implemented your suggestion.

And I also tested without it. Using maximized windows and floating windows.
No issues occurred.

You said you were different from me in the sense that you looked at the code and saw what made sense or not or could create regressions from it.

Well. I tested. And it looks like you were wrong about it creating regressions. Using the layouts I didn't test previously.

So maybe looking at the code sometimes in insufficient. Testing also matters.

**Review by manuelnogueiramov — DISMISSED** [review]
Looks great.

Good work! :tada:

**Review by diasnad — DISMISSED** [review]

**Review by diasnad — APPROVED** [review]
